### ISSUE TITLE
YaruCompactLayout style isn't consistent with the rest of the UI

### ISSUE THREAD

**Issue by Jupi007** [ubuntu/yaru_widgets.dart#175]
First, the selected icon is using the accent color, while tabbed page isn't:

![2022-08-07_17-32](https://user-images.githubusercontent.com/36476595/183299459-c32930f0-68a2-4ada-81fe-729a34a3d84e.png)

Secondly, the selection background isn't wrapping the label, and the hover background isn't fitting the full width, btw it also uses the accent with opacity on hover, but grey on click and this blends bad:

![Capture d’écran du 2022-08-07 17-34-25](https://user-images.githubusercontent.com/36476595/183299457-e2cde158-9a64-4345-bd38-2e902a445bfd.png)

I tried to take a look into the code, but I think that the `NavigationRail` customization is very limited.

**Comment by Feichtmeier** [comment]
Yes navi rail theming is very limited. Since navi rail provides many nice things I would like to stick to the navirail. Since tabs and the navigation rail are two different UI elements, I don't think they need to match 100%

**Comment by Feichtmeier** [comment]
![grafik](https://user-images.githubusercontent.com/15329494/183379259-91b06897-e63c-4c63-aa23-41867c1b1904.png)
![grafik](https://user-images.githubusercontent.com/15329494/183379369-085df54b-ad03-499b-94c6-516f40a505fe.png)

Changing the indicator color for the selected item and the text color for the unselected item is possible though

What triggers me the most is the primary color square on hover and clicked. I personally like the fact that the indicator background does not include the text. But it is not possible to change this. Unless, ofc, we create our own navigatrion rail 🤷 but we already have the regular not compact layout. The compact layout can be made even more compact if you chose NavigationRailLabelType.none (which is the default). The point of this layout was actually to have a very compact layout. Always showing the label for the software app was a feedback from Elio

**Comment by Jupi007** [comment]
Honestly, the selected backgound which is not covering the label is not a real "problem" and we can leave it as it.
IMO that is very problematic is the hover color, and the fact that it isn't covering the full width on hover, if the label is too short.

**Comment by Feichtmeier** [comment]
We could try to replace the navirail or the YaruWideLayout with a modified or parameterized version of YaruLandscapeLayout :) Then we should be independent from the material3 fun

**Comment by Feichtmeier** [comment]
I really like this compact look TBH
![grafik](https://user-images.githubusercontent.com/15329494/183403963-a9cea85d-8c2e-4ea6-919c-f507351bec85.png)
So the replacement should look like this (minus the colored icon and minus the weird hover/splash)

**Comment by Jupi007** [comment]
I also like this compact rail look :)

I agree that replace the navrail with a custom widget that we could style as we want is the best idea. I also was thinking about reuse the `YaruLandscapeLayout` in a custom way.

P.S.: I think you commented twice ;)

**Comment by Feichtmeier** [comment]
I think we should create our own navigationrail for now, personally I don't have the time to go through the long process of getting this into flutter itself
